### ISSUE TITLE
[Bug Report] 4D boolean keep-masks produce incorrect TransformerBridge causal loss

### ISSUE THREAD

**Issue by emerardd** [TransformerLensOrg/TransformerLens#1807]
**Describe the bug**

`TransformerBridge(..., return_type="loss" | "both")` can return a wrong loss or `NaN` when an architecture uses a 4D boolean attention mask with `True` meaning *visible*. The logits are correct. `BridgeCore._prepare_loss_attention_mask()` instead interprets every 4D boolean mask as `True` meaning *blocked*, so it selects the wrong next-token targets. Explicit `labels` use the same reduction path.

**Code example**

With a randomly initialized tiny GPT-2 (`GPT2Config(vocab_size=32, n_positions=8, n_embd=16, n_layer=1, n_head=2, _attn_implementation="eager")`) wrapped by `build_bridge_from_module()`, pass tokens `[[1, 2, 3, 4]]` using either a 2D all-ones mask or the equivalent 4D lower-triangular boolean keep-mask. At seed 1:

```text
max absolute logit difference: 0.0
2D loss:                    3.5207941532
4D boolean keep-mask loss:  NaN
```

With `[[1, 2, 3, 0, 0]]` and the last two positions masked, the losses are `3.5178113` (2D) versus `3.3149614` (4D). This is reproducible without downloading a model.

**Expected behavior**

Equivalent masks should score the same unmasked next-token transitions. Mask polarity must follow the architecture's forward-attention contract: GPT-2's 4D boolean mask uses `True=visible`, while the TL-native model uses `True=blocked`. A blanket inversion would break the native path.

**System Info**

Source checkout of `dev` at `2e4cdd12`, Windows, Python 3.12.10; tiny models initialized locally with the repository's `uv` environment.

**Additional context**

The existing native 4D-mask loss test covers the opposite boolean convention. A fix should test both conventions, padded and unpadded inputs, `return_type="loss"`/`"both"`, and explicit labels.

### Checklist

- [x] I checked open and closed Issues and PRs for equivalent 4D mask, loss, and mask-polarity reports.

**Comment by emerardd** [comment]
I can work on this. I reproduced the loss mismatch with a tiny, locally initialized GPT-2 and will submit a focused fix with regressions for both the GPT-2 keep-mask convention and the native blocked-mask convention, including explicit labels.
